app/src/components/generative/ModelMenu.tsx

```diff
@@ -191,59 +325,127 @@ type ModelsByProviderMenuProps = {
 function ModelsByProviderMenu({
   modelsByProvider,
   providerInfoMap,
+  customProviders,
+  defaultModelByProvider,
   onChange,
 }: ModelsByProviderMenuProps) {
-  const handleModelSelect = (providerKey: string, modelName: string) => {
+  const handleModelSelect = (
+    providerKey: string,
+    modelName: string,
+    customProviderId?: string
+  ) => {
     if (isModelProvider(providerKey)) {
       onChange?.({
         provider: providerKey,
         modelName,
+        customProviderId,
       });
     }
   };
 
+  const hasBuiltInResults = modelsByProvider.size > 0;
+  const hasCustomResults = customProviders.length > 0;
+  const hasResults = hasBuiltInResults || hasCustomResults;
+
   return (
     <Menu
       css={menuWidthCSS}
       autoFocus={false}
       onAction={(key) => {
-        const keyStr = String(key);
-        // Key format is "providerKey:modelName"
-        const separatorIndex = keyStr.indexOf(":");
-        if (separatorIndex > 0) {
-          const providerKey = keyStr.slice(0, separatorIndex);
-          const modelName = keyStr.slice(separatorIndex + 1);
-          handleModelSelect(providerKey, modelName);
+        const parsed = decodeMenuKey(String(key));
```

Comment:
nit: parsed as a variable name is abstract. Why not modelInfo or something like that. Because to me the decoded value is the model designaiton right?